man/man7/zfsprops.7

```diff
@@ -900,13 +900,16 @@ zeroes (the NUL byte).
 When a zero-filled block is detected, it is stored as
 a hole and not compressed using the indicated compression algorithm.
 .Pp
-Any block being compressed must be no larger than 7/8 of its original size
-after compression, otherwise the compression will not be considered worthwhile
-and the block saved uncompressed.
-Note that when the logical block is less than
-8 times the disk sector size this effectively reduces the necessary compression
-ratio; for example, 8kB blocks on disks with 4kB disk sectors must compress to 1/2
-or less of their original size.
+All blocks are allocated as a whole number of sectors
+(chunks of 2^ashift bytes, e.g. 512B or 4KB).
+Compression may result in a non-sector-aligned size, which will be rounded up
+to a whole number of sectors.
+If compression saves less than one whole sector,
+the block will be stored uncompressed.
+Therefore, blocks whose logical size is a small number of sectors will
+experience less compression ratio
+(e.g. for recordsize=16K with 4K sectors, which have 4 sectors per block,
```

Comment:
```
(e.g. for
.Sy recordsize Ns = Ns Sy 16K
with 4K sectors, ...
```

Reply:
Thank you, used your examples as a base.